clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -788,6 +801,19 @@ public void handleResponse(AbstractResponse response) {
                 result.done();
             }
         }
+
+        @Override
+        public long retryBackoffMs() {
+            return this.retryBackoffMs;
+        }
+
+        private void maybeOverrideRetryBackoffMs() {
+            // We only want to reduce the backoff when retrying the first AddPartition which errored out due to a
+            // CONCURRENT_TRANSACTIONS error since this means that the previous transaction is still completeing and
+            // we don't want to wait too long before trying to start the new one.
```

Comment:
`completeing` -> `completing`

We should also probably mention that this is a temporary solution. And we should file a JIRA for the proper solution.